test/e2e/persistent_volumes.go

```diff
@@ -327,241 +411,396 @@ func createWaitAndDeletePod(f *framework.Framework, c *client.Client, ns string,
+			})
+		})
 	})
-
-	// Create a nfs PV first, then a pre-bound PVC that matches the PV,
-	// and a pod that contains the claim. Verify that the PV and PVC bind
-	// correctly, and that the pod can write to the nfs volume.
-	It("create a PV and a pre-bound PVC: test write access [Flaky]", func() {
-
-		pv, pvc, err = createPVPVC(c, serverIP, ns, true)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
-
-		// validate PV-PVC, create and verify writer pod, delete PVC
-		// and PV
-		pv, pvc, err = completeTest(f, c, ns, pv, pvc)
-		if err != nil {
-			framework.Failf("%v", err)
-		}
+	///////////////////////////////////////////////////////////////////////
+	//				GCE PD
+	///////////////////////////////////////////////////////////////////////
+	// Testing configurations of single a PV/PVC pair attached to a GCE PD
+	framework.KubeDescribe("PV:GCEPD", func() {
+
+		var (
+			diskName  string
+			err       error
+			pv        *api.PersistentVolume
+			pvc       *api.PersistentVolumeClaim
+			clientPod *api.Pod
+			pvConfig  persistentVolumeConfig
+		)
+
+		BeforeEach(func() {
+			framework.SkipUnlessProviderIs("gce")
+			if diskName == "" {
+				diskName, err = createPDWithRetry()
+				Expect(err).NotTo(HaveOccurred())
+				pvConfig = persistentVolumeConfig{
+					namePrefix: "gce-",
+					pvSource: api.PersistentVolumeSource{
+						GCEPersistentDisk: &api.GCEPersistentDiskVolumeSource{
+							PDName:   diskName,
+							FSType:   "ext3",
+							ReadOnly: false,
+						},
+					},
+					prebind: nil,
+				}
+			}
+		})
+
+		AfterEach(func() {
+			if c != nil {
+				if clientPod != nil && len(clientPod.Name) > 0 {
+					_, err := c.Pods(ns).Get(clientPod.Name)
+					if !apierrs.IsNotFound(err) {
+						Expect(err).NotTo(HaveOccurred())
+					        framework.Logf("AfterEach: deleting client Pod %v", clientPod.Name)
+        					err = c.Pods(ns).Delete(clientPod.Name, nil)
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleted client Pod %v", clientPod.Name)
+					}
+				}
+				clientPod = nil
+				if pvc != nil && len(pvc.Name) > 0 {
+					_, err := c.PersistentVolumeClaims(ns).Get(pvc.Name)
+					if !apierrs.IsNotFound(err) {
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleting PVC %v", pvc.Name)
+						err = c.PersistentVolumeClaims(ns).Delete(pvc.Name)
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleted PVC %v", pvc.Name)
+					}
+				}
+				pvc = nil
+				if pv != nil && len(pv.Name) > 0 {
+					_, err := c.PersistentVolumes().Get(pv.Name)
+					if !apierrs.IsNotFound(err) {
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleting PV %v", pv.Name)
+						err := c.PersistentVolumes().Delete(pv.Name)
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("AfterEach: deleted PV %v", pv.Name)
+					}
+				}
+				pv = nil
+			}
+		})
+
+		AddCleanupAction(func() {
+			if len(diskName) > 0 {
```

Comment:
Sorry, I'm not following your meaning here.  What do you mean by "moved to deletePDwithRetry()"?